app/services/gsuite.rb

```diff
@@ -0,0 +1,15 @@
+require 'http'
+
+module Gsuite
+  class DomainLookupError < StandardError; end
+
+  def self.is_gsuite_domain?(domain)
+    response = HTTP.get("https://dns.google.com/resolve?name=#{domain}&type=MX")
+    if response.status.success?
+      payload = JSON.parse(response.body)
+      payload['Answer']&.any? { |h| h['data'] =~ /\.google\.com/i }
```

Comment:
Should also look out for older gsuite domains for which the MX records point to googlemail.com

Reply:
Good call, will add that in now